### DISCUSSION TITLE
Ghelper doesn't run at logon

### DISCUSSION THREAD

**Discussion by ghost** [seerge/g-helper#1726] [Q&A]
Thank you for great tool, works great so far. I have issue of it not running at windows log on. I have kept the battery limter setting to 60% and still, the laptop starts charging for like 2 minutes on startup and then ghelper activates and charging stops. 

because of startiong and hibernating, my laptop has charged to 80% bit by bit everyday from 60%. 

I have applied the following fixes so far from reading the discussions and posts here:

1) uncheck the startup box and check it again
2) stop asus services and only asus HID control service is active
3) keep windows power plan to balanced
4) apply "run with or without user logon" and "run with highest privilegdes"
5) remove the task manually from task scheduler and tick again on the startup box in ghelper
6) run ghelper as administrator after clean restart

 My laptop : ROG G14 401IV 4900 HS
Ghelper version : 0.134
Windows 11 23H2

I have read the following discussions already and then came here:

https://github.com/seerge/g-helper/issues/1079
https://github.com/seerge/g-helper/issues/630
https://github.com/seerge/g-helper/issues/43
https://github.com/seerge/g-helper/issues/29
https://github.com/seerge/g-helper/discussions/742

If anyone can help, please. because the laptop keeps charging minute per minute everyday and reached 80-90%. Note that ghelper starts after some time when I have already unlocked the screen.

**Comment by seerge** [comment]
@dev1312000 unfortunately this is how apps under windows work :) G-Helper is an app with a UI (forms) and it has to be run under some user, so it can't start before you actually log in.

If the only reason you want to start app earlier is charging limiter, you can set a limit as a power-shell command and set it in task manager instead. Command needs to run with administrator privileges 

```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00120057, 60)
```

Or you can create a .bat file and set it to run on startup

```
powershell -Command "(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00120057, 60)"

**Reply by ghost** [reply]
Correct, I understand. But yesterday I also started the asus services and run MYASUS app with battery limiter set to 60%. and on the restart, laptop charging light was orange only for 10 seconds and instantly turned white even before log on (unlocked screen). What setting in MYASUS works that way but in Ghelper doesn't? after a week or so I have to discharge my laptop to 58% again to use it at 60. 

the fixes provided by you are permanent? or I can charge to 100% in ghelper setting even after those commands are active? I am not a coding students so asking dumb questions please forgive me. Can I revert back to charging 80 or 100% when I need it? If yes,then I will run this command.

Thank you for your involvement.

**Reply by seerge** [reply]
@dev1312000 MyASUS relies on 9 asus services (that can start w/o user), and g-helper doesn't cause it tries to be simplier and lighter, right ? :)

I think I'v offered you an option of how to set limit before login. That's everything I can do

**Reply by ghost** [reply]
Yes I Understood clearly. I ran the command with administrator priviledges powershell but it showed this :
![JKdB5TFlQu](https://github.com/seerge/g-helper/assets/76037097/2223f537-eb42-4cb3-815d-9f6c8ca0ff58)

And as second option I ran the bat file but didn't seema to work. Is it correctly written? I am sorry to bother I don't have connection with coding at all.

**Reply by seerge** [reply]
I think you didn’t run command from powershell with admin rights :) access denied from windows is exactly about that

**Reply by ghost** [reply]
Let me do it again and come back. thanks 👍

**Reply by ghost** [reply]
Take a look. I ran as an administrator in the latest powershell version 7.4.0
![image](https://github.com/seerge/g-helper/assets/76037097/eb1aa5c4-9663-407e-accc-e1bf747af66b)

**Reply by seerge** [reply]
Then, probably, on your model asus control interface doesn’t expose this endpoints in powershell :)

I’m afraid i can’t help anyhow

**Reply by ghost** [reply]
It's okay. Your support till here was more than good. Thank you for the help. If anyone reading this discussion knows what to do, please acknowledge me. Thank you.

**Comment by Kikimahe** [comment]
> ...  the laptop starts charging for like 2 minutes on startup and then ghelper activates and charging stops.

Hi, 

2 minutes to start Windows ? 
Isn't it the time **to restart** Windows instead? ;-)

If you confirm that it 's 2min to start Windows, it is normal that the battery charges very quickly until Ghelper start.

You should then try to optimize Windows loading time by deactivating as many software as possible at startup (task manager, task scheduler, autoruns64 ....)

My laptop boots up in 15s on Windows.
- 5 seconds for BIOS (Asus logo)
- 10 seconds for Windows.

But the restart takes much longer and therefore the battery has more time to charge.

**Reply by ghost** [reply]
Yes my bad. It is restart time. But sometimes I have to restart because of either antivirus or my engineering software problem, and ghelper starts charging to 1.5% or 2%. I have to restart every two days a week to keep it fresh.

I only use three engineering softwares, that's it. and my windows startup time is faster to 12s but my problem is, even after logon, Ghelper loads like 20s later. I can't figure that out also. in my tray Icon, Nvidia setting, Microsoft seurity and my antivirus all 3 starts early, but gelper shows itself after long time than them. what can I do about that?

**Reply by Kikimahe** [reply]
>Ghelper loads like 20s later

Oo.
You're right, 20 seconds is unusually long...

1) Wait a minute, you have 2 anti-viruses? Microsoft security + another one? 
But why ? :-)  Maybe this is the problem ...
Try to unistall the second one et keep Microsoft security. 

The best anti-virus is located between the chair and the screen. ;-)


2) Else, check if there is any software or service that is causing Ghelper to run slowly. 
- Try disabling all non-Microsoft services in MSCONFIG.
- Try disabling all software at startup in TASK MANAGER.
And test Windows startup

3) You can also check in the Ghelper task that the counter is at 1s and not at 10 or 20s
![Capture d'écran 2023-12-12 150934](https://github.com/seerge/g-helper/assets/142881210/f52d4d0c-a749-4d71-adf9-d34306ce38e3)


And maybe modify the Ghelper task start trigger.

![Sans titre-1](https://github.com/seerge/g-helper/assets/142881210/89d6b39d-e4ed-47b8-b93f-4ee06b9022fa)


Sorry for my french screen. ;'(

**Reply by ghost** [reply]
No, Microsoft defender already comes installed but I need to install an antivirus to protect when operating this big softwares because they may crash the system if a plugin is faulty. Sure I will try out these options. Thank you.

**Comment by MC-dusk** [comment]
see https://github.com/seerge/g-helper/discussions/1773#discussioncomment-8480920

**Comment by JuneXIE001** [comment]
I encountered the same situation, but when using MYASUS, it won't charge when restarting. I hope to turn off this feature of G-helper and use MYASUS. Can you please help me find a way to turn off this feature of G-helper
